pioneer/packages/joy-tokenomics/src/index.tsx

```diff
@@ -1,33 +1,37 @@
 import React from 'react';
-import translate from './translate';
+import { useTranslation } from './translate';
 import { Route, Switch } from 'react-router';
 import { Tabs } from '@polkadot/react-components';
 import Overview from './Overview';
-
 import { AppProps, I18nProps } from '@polkadot/react-components/types';
+import { TransportProvider } from '@polkadot/joy-utils/react/context';
 
 interface Props extends AppProps, I18nProps {}
 
-function App ({ basePath, t }: Props): React.ReactElement<Props> {
+function App ({ basePath }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+
   return (
-    <main>
-      <header>
-        <Tabs
-          basePath={basePath}
-          items={[
-            {
-              isRoot: true,
-              name: 'overview',
-              text: t('Tokenomics')
-            }
-          ]}
-        />
-      </header>
-      <Switch>
-        <Route component={Overview} />
-      </Switch>
-    </main>
+    <TransportProvider>
+      <main>
+        <header>
+          <Tabs
+            basePath={basePath}
+            items={[
+              {
+                isRoot: true,
+                name: 'overview',
+                text: t('Tokenomics')
+              }
+            ]}
+          />
+        </header>
+        <Switch>
+          <Route component={Overview} />
+        </Switch>
+      </main>
+    </TransportProvider>
   );
 }
 
-export default translate(App);
```

Comment:
That's the change that caused `TypeScript` error in `packages/apps-routing/src/joy-tokenomics.ts`.
A simple way to fix it:
- Revert this (along with `pioneer/packages/joy-tokenomics/src/translate.ts` changes)
- Change Props interface to:
```
import { RouteProps as AppMainRouteProps } from '@polkadot/apps-routing/types';

interface Props extends AppMainRouteProps, I18nProps {}
```

Reply:
The reason I initially changed this was because the error was immediately there after moving to Iznik but I couldn't figure out what it was and just left it like that. 

I tried reverting the changes and doing exactly like you wrote but it still didn't work. Same error. But just adding these two lines to the file as it is now allows it to work properly. Should I just go along with that, then?

Reply:
Still getting an error. While hovering translate(App):
```
Types of parameters '__0' and 'props' are incompatible.
      Type 'PropsWithChildren<WithTranslation>' is missing the following properties from type 'Props': basePath, onStatusChange
```
While hovering Component in routing:
```
Type 'ComponentType<Pick<WithTranslation, never> & WithTranslationProps>' is not assignable to type 'ComponentType<RouteProps>'.
```
I got it working by not reverting the changes and just deleting and not extending I18nProps. I assume that is far from optimal but I thought it'd be good to mention.

Reply:
Taking another look at it I think both my and your solution works, because `withTranslation(['joy-tokenomics'])(App)` should make `I18nProps` optional in the resulting component (I'm not sure why it caused error in your IDE).

If you're using hooks instead of `withTranslation` then just removing `I18nProps` from `Props` seems perfectly valid and is probably even better, so if it works - go for it!